### DISCUSSION TITLE
Golang Serverless Function, Help With Paths

### DISCUSSION THREAD

**Discussion by KnlnKS** [vercel/community#447] [Help] [answered]
Hi, I'm trying to use an HTML Template with a Go Serverless function, however I'm running into an issue. I get this error:
```
[GET] /api?username=KnlnKS
15:33:24:25
open /vercel/path1/templates/submissionData.svg: no such file or directory: PathError
```
The issue seems to be this bit of code:
```go
func GetSubmissionStatsTemplate() *template.Template {
	tmpl, err := template.ParseFiles("/vercel/templates/submissionData.svg")
	if err != nil {
		panic(err)
	}
	return tmpl
}
```
The code works while using the Vercel CLI, but fails when deployed. Any ideas on how to remedy this?

**Comment by samcx** [comment]
This Support Article should help → https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions! Let us know if this resolves your issue.

**Reply by irohitb** [reply]
@samcx this is for NodeJS, the qustion is for golang

**Reply by KnlnKS** [reply, marked as answer]
I did eventually find a workaround. I just ended up embedding the file.

```go
//go:embed submissionStats.svg
var submissionStatsSVG string
var submissionStatsTmpl *template.Template = nil

func GetSubmissionStatsTemplate() *template.Template {
	if submissionStatsTmpl == nil {
		submissionStatsTmpl, _ = template.New("SubmissionStats").Parse(`{{define "T"}}` + submissionStatsSVG + `{{end}}`)
	}
	return submissionStatsTmpl
}
```

**Reply by samcx** [reply]
@irohitb My apologies! Have you tried using @KnlnKS workaround?
